### ISSUE TITLE
negative ignore_index not working

### ISSUE THREAD

**Issue by stefan-schroedl** [Lightning-AI/metrics#1393]
## 🐛 Bug
I would like to use metrics (e.g., Precision) with an ignore_index that is not a regular class index (e.g., ignore_index = -1).
Trying this with the code below results in an error.
I was expecting this to be possible. Maybe the code is not designed to have this feature? In this case, it would be good to update the documentation to clarify that the ignore_index has to be between 0 and num_classes-1.

Thank you!

### To Reproduce

Running below code sample results in an "index out of bounds" error.

#### Code sample

```python
import torch
from torchmetrics.classification import Precision
preds  = torch.tensor( [0, 1, 0, 3, 0, 2, 3, 1, 0, 3])
target  = torch.tensor([0, 1, 0, 3, -1, 0, 0, 1, 2, 3])
num_classes=4
prec = Precision(num_classes=num_classes, average="macro", ignore_index=-1)
print(prec(preds, target))
```

### Expected behavior

Expected result of 0.5833

### Environment

Installed with `pip install torchmetrics==0.9.3`

Linux environment, python 3.9.10

**Comment by SkafteNicki** [comment]
Hi @stefan-schroedl,
Please update to v10.0 of torchmetrics, where the feature to set `ignore_index` to a negative number was implemented. Only change then necessary to your code is adding the `task` argument when initializing the metric:

```python
import torch
from torchmetrics.classification import Precision
preds  = torch.tensor( [0, 1, 0, 3, 0, 2, 3, 1, 0, 3])
target  = torch.tensor([0, 1, 0, 3, -1, 0, 0, 1, 2, 3])
num_classes=4
prec = Precision(task='multiclass', num_classes=num_classes, average="macro", ignore_index=-1)
print(prec(preds, target))
# prints tensor(0.5833) as expected
```
Closing issue.
